docs/photonics/examples/SFWM_Aeff.py

```diff
@@ -0,0 +1,160 @@
+from skfem.helpers import dot
+from skfem.io.meshio import from_meshio
+
+from femwell.maxwell.waveguide import compute_modes
+from femwell.mesh import mesh_from_OrderedDict
+
+
+def calculate_sfwm_Aeff(basis: Basis, mode_p, mode_s, mode_i) -> np.complex64:
+    """
+    Calculates the overlap integral for SFWM process by considering the interactions
+    between pump, signal, and idler modes in the xy plane.
+
+    Args:
+        basis (Basis): Common basis used for all modes in the same geometric structure.
+        mode_p, mode_s, mode_i: Mode instances for pump, signal, and idler, respectively.
+
+    Returns:
+        np.complex64: The Aeff result for the SFWM process(1/overlap integral).
+    """
+
+    def normalization_factor_mode(mode):
+        @Functional
+        def E2(w):
+            return dot(w["E"][0], np.conj(w["E"][0]))
+
+        E = mode.basis.interpolate(mode.E)  # [0]=xy [1]=z
+        E_squared_integral = E2.assemble(mode.basis, E=E)
+        normalization_factor = 1 / np.sqrt(E_squared_integral)
+        return normalization_factor  # Return the normalization factor instead of modifying the mode
+
+    # Apply normalization factors to the electric fields for the overlap calculation
+    @Functional(dtype=np.complex64)
+    def sfwm_overlap(w):
+
+        E_p_xy = w["E_p"][0]  # shape: (2, x, 3)
+        E_s_xy = w["E_s"][0]  # shape: (2, x, 3)
+        E_i_xy = w["E_i"][0]  # shape: (2, x, 3)
+        
+        overlap_Ex = E_p_xy[0,:,0]*E_p_xy[0,:,0]*np.conj(E_s_xy[0,:,0])*np.conj(E_i_xy[0,:,0]) + \
+                     E_p_xy[1,:,0]*E_p_xy[1,:,0]*np.conj(E_s_xy[1,:,0])*np.conj(E_i_xy[1,:,0])
+        overlap_Ey = E_p_xy[0,:,1]*E_p_xy[0,:,1]*np.conj(E_s_xy[0,:,1])*np.conj(E_i_xy[0,:,1]) + \
+                     E_p_xy[1,:,1]*E_p_xy[1,:,1]*np.conj(E_s_xy[1,:,1])*np.conj(E_i_xy[1,:,1])
+        overlap_Ez = E_p_xy[0,:,2]*E_p_xy[0,:,2]*np.conj(E_s_xy[0,:,2])*np.conj(E_i_xy[0,:,2]) + \
+                     E_p_xy[1,:,2]*E_p_xy[1,:,2]*np.conj(E_s_xy[1,:,2])*np.conj(E_i_xy[1,:,2])
+    
+        return np.array([overlap_Ex, overlap_Ey, overlap_Ez]).T
+ 
+        #return dot(w["E_p"][0], w["E_p"][0]) * dot(np.conj(w["E_s"][0]), np.conj(w["E_i"][0]))#?
+        #return dot(w["E_p"][0], np.conj(w["E_s"][0])) * dot(w["E_p"][0], np.conj(w["E_i"][0]))#??
+ 
+    overlap_result = sfwm_overlap.assemble(
+        basis,
+        E_p=mode_p.basis.interpolate(mode_p.E * normalization_factor_mode(mode_p)),
+        E_s=mode_s.basis.interpolate(mode_s.E * normalization_factor_mode(mode_s)),
+        E_i=mode_i.basis.interpolate(mode_i.E * normalization_factor_mode(mode_i)),
+    )
+
+    return 1 / overlap_result
+
+
+# Dispersion relations of materials
+# Core
+def n_X(wavelength):
+    x = wavelength
+    return (
+        1
+        + 2.19244563 / (1 - (0.20746607 / x) ** 2)
+        + 0.13435116 / (1 - (0.3985835 / x) ** 2)
+        + 2.20997784 / (1 - (0.20747044 / x) ** 2)
+    ) ** 0.5
```

Comment:
I'd guess we should use here
```suggestion
def n_X(wavelength):
    x = wavelength * 1000
    return (
        1
        + 3.0249 / (1 - (135.3406 / x) ** 2)
        + 40314 / (1 - (1239842 / x) ** 2)
    ) ** 0.5

```

Then it matches way better the linked example, there they got 0.59, I get 0.6 :)
(And we should probably call it n_silicon_nitride)